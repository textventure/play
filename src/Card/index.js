import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';

const styles = theme => ({
  paper: {
    margin: theme.spacing.unit * 2, // 16px
    padding: theme.spacing.unit * 3, // 24px
    whiteSpace: 'pre-line', // respect newlines in text
  },
});

class Card extends Component {
  render() {
    const { children, classes, className } = this.props;
    return (
      <Paper className={[classes.paper, className].join(' ')}>{children}</Paper>
    );
  }
}

export default withStyles(styles)(Card);
