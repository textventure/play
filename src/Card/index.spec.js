import React from 'react';
import { shallow } from 'enzyme';
import Card from '.';

let wrapper;

describe('when props.children="foo"', () => {
  beforeAll(() => {
    wrapper = shallow(<Card>foo</Card>).dive();
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});

describe('when props.className="foo"', () => {
  beforeAll(() => {
    wrapper = shallow(<Card className="foo" />).dive();
  });

  it('renders correctly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
