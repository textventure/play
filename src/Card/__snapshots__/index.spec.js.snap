// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when props.children="foo" renders correctly 1`] = `
<WithStyles(Paper)
  className="Card-paper-1 "
>
  foo
</WithStyles(Paper)>
`;

exports[`when props.className="foo" renders correctly 1`] = `
<WithStyles(Paper)
  className="Card-paper-1 foo"
/>
`;
